Modules/_collectionsmodule.c

```diff
@@ -1134,36 +1168,30 @@ deque_insert(dequeobject *deque, PyObject *const *args, Py_ssize_t nargs)
 PyDoc_STRVAR(insert_doc,
 "D.insert(index, object) -- insert object before index");
 
+static int deque_del_item(dequeobject *, Py_ssize_t);
+
 static PyObject *
 deque_remove(dequeobject *deque, PyObject *value)
 {
-    Py_ssize_t i, n=Py_SIZE(deque);
+    Py_ssize_t i;
+    size_t start_state = deque->state;
+    int rv;
 
-    for (i=0 ; i<n ; i++) {
-        PyObject *item = deque->leftblock->data[deque->leftindex];
-        int cmp = PyObject_RichCompareBool(item, value, Py_EQ);
+    i = _deque_index(deque, value, 0, Py_SIZE(deque));
+    if (i == -1 && PyErr_Occurred()) {
```

Comment:
Since `-1` always indicates an error is set, you can skip the ` && PyErr_Occurred()` and instead put an assertion inside the if-body.